semgrep_output_v1.atd

```diff
@@ -811,13 +811,15 @@ type found_dependency <ocaml attr="deriving show"> = {
 type api_scans_findings <ocaml attr="deriving show"> = {
   (* TODO: ?version: version option; *)
    findings: finding list;
+   ignores: finding list;   
    token: string nullable;
-   gitlab_token: string nullable;
    searched_paths: string list;
-   (* TODO: rule_id list *)
+   renamed_paths: string list;
```

Comment:
Do we need to double check this field? It looks like it was moved around a bit but I think it's a no-op.
```
    ⋮ 301│        api_scans_findings = out.ApiScansFindings(
 282⋮ 302│            # send a backup token in case the app is not available
 283⋮    │            "token": os.getenv("GITHUB_TOKEN"),
 284⋮    │            "gitlab_token": os.getenv("GITLAB_TOKEN"),
 285⋮    │            "findings": findings,
 286⋮    │            "searched_paths": [str(t) for t in targets],
 287⋮    │            "renamed_paths": [str(rt) for rt in renamed_targets],
 288⋮    │            "rule_ids": rule_ids,
 289⋮    │            "cai_ids": cai_ids,
    ⋮ 303│            token=token,
    ⋮ 304│            findings=findings,
    ⋮ 305│            searched_paths=[str(t) for t in sorted(targets)],
    ⋮ 306│            rule_ids=rule_ids,
    ⋮ 307│            gitlab_token=None,
    ⋮ 308│        )
    ⋮ 309│        # TODO: add those fields in semgrep_output_v1.atd spec
    ⋮ 310│        findings_and_ignores = {
    ⋮ 311│            **api_scans_findings.to_json(),
    ⋮ 312│            "renamed_paths": [str(rt) for rt in sorted(renamed_targets)],
 290⋮ 313│            "ignores": ignores,
 291⋮ 314│        }
```

Reply:
This field has been included in the json sent to `findings_and_ignores` since at least 0.123.0: https://github.com/returntocorp/semgrep/blob/084ff5f16866ab66337655dbc06672e05d3089dc/cli/src/semgrep/app/scans.py#L281

It just hasn't been included in the ATD type

Reply:
Discussed in slack, and this is fine

Reply:
jonas got this diff backwards. Here's the code creating this field in 0.123.0: https://github.com/returntocorp/semgrep/blob/084ff5f16866ab66337655dbc06672e05d3089dc/cli/src/semgrep/app/scans.py#L281

It's hasn't changed type/existence since then, so this change should indeed be a no-op, but will enable us to do more type-checking

Reply:
cc @aryx, should be good to merge if you approve